### PR TITLE
Allow bulk move/copy actions to happen in the background

### PR DESCRIPTION
**Pull request by oliverguenther**
https://community.openproject.org/wp/46476

### CODE REVIEW

File: app/controllers/work_packages/moves_controller.rb [hunk 1]
```diff
@@ -40,45 +40,53 @@ def new
...
-              WorkPackages::Bulk::CopyService
-            else
-              WorkPackages::Bulk::MoveService
-            end
-
-    klass
-      .new(user: current_user, work_packages: @work_packages)
-      .call(attributes_for_create)
+  def perform_operation
+    if @work_packages.count <= Setting.work_packages_bulk_request_limit
```
Comment (machisuji) [inline app/controllers/work_packages/moves_controller.rb]: I think this is only the selected work package to be copied. This count would not include all the descendants, would it? In In the case that caused the bug report only a single work package had been selected to be copied. But it had 1000+ descendants.

Comment (oliverguenther) [reply app/controllers/work_packages/moves_controller.rb]: Good point, fixed

**Comment by machisuji** [comment]
Looks good and works in principle. Just the error handling doesn't look quite right yet. For example:

image

**Comment by oliverguenther** [comment]
Good catch @machisuji , I missed a spec for that part. It should print the correct output now

**Review by machisuji — APPROVED** [review]
👍

**Pull request merged**